src/components/MaterialUI/componentProps.test.js

```diff
@@ -0,0 +1,63 @@
+import ComponentProps from "./componentProps";
+
+describe("Component Props", () => {
+	it("Sets the prop value", () => {
+		var componentProps = new ComponentProps();
+
+		componentProps.componentProps.set("key1", null);
+
+		componentProps.set("key1", "value1");
+
+		expect(componentProps.componentProps.get("key1"), "to equal", "value1");
+	});
+
+	it("Throws if prop key is missing", () => {
+		var componentProps = new ComponentProps();
```

Comment:
const, const...

Reply:
why const for a local variable? consts should be used for places where variable shouldn't be overwritten by design. In this case I don't care about it there aren't such requirements.

Reply:
Replaced all the var with let